client/verta/tests/test_organization.py

```diff
@@ -24,9 +26,75 @@ def test_create_msg():
     def test_create(self, client):
         name = _utils.generate_default_name()
         org = client._create_organization(name)
         assert org
         assert org.id == client._get_organization(name).id
+
+        org.delete()
+
+    def test_create_same_name_diff_workspace(self, client, organization, in_tempdir, created_endpoints, created_registered_models, created_datasets):
+        # creating some entities:
+        project_name = _utils.generate_default_name()
+        exp_name = _utils.generate_default_name()
+        run_name = _utils.generate_default_name()
+        dataset_name = _utils.generate_default_name()
+        repository_name = _utils.generate_default_name()
+        model_name = _utils.generate_default_name()
+        version_name = _utils.generate_default_name()
+        endpoint_path = _utils.generate_default_name()
+
+        project = client.create_project(project_name)
+        exp = client.create_experiment(exp_name)
+        run = client.create_experiment_run(run_name)
+        repository = client.get_or_create_repository(name=repository_name)
+
+        dataset = client._create_dataset2(dataset_name)
+        created_datasets.append(dataset)
+        model = client.create_registered_model(name=model_name)
+        version = model.create_version(name=version_name)
+        created_registered_models.append(model)
+
+        endpoint = client.create_endpoint(path=endpoint_path)
+        created_endpoints.append(endpoint)
+
+        # Switching workspace name:
+        client_config = {
+            "workspace": organization.name
+        }
+
+        filepath = "verta_config.json"
+        with open(filepath, "w") as f:
+            json.dump(client_config, f)
+
+        client = Client()
+
+        # create entities with same name, but different workspace:
+        new_model = client.create_registered_model(name=model_name)
+        new_version = new_model.create_version(name=version_name)
+        new_endpoint = client.create_endpoint(path=endpoint_path)
+
+        new_project = client.create_project(project_name)
+        new_exp = client.create_experiment(exp_name)
+        new_run = client.create_experiment_run(run_name)
+        new_repository = client.get_or_create_repository(name=repository_name, workspace=organization.name)
+
+        new_dataset = client._create_dataset2(dataset_name)
+        created_datasets.append(new_dataset)
+
+        created_endpoints.append(new_endpoint)
+        created_registered_models.append(new_model)
+
+        assert model.id != new_model.id
+        assert version.id != new_version.id
+        assert endpoint.id != new_endpoint.id
+        assert project.id != new_project.id
+        assert exp.id != new_exp.id
+        assert run.id != new_run.id
+        assert dataset.id != new_dataset.id
+        assert repository.id != new_repository.id
+
+        project.delete()  # have to delete manually because creating dataset makes project untracked by client context.
```

Comment:
> have to delete manually because creating dataset makes project untracked by client context.
Does it really?? That seems odd. Could you do me a favor and point out where in the client code that happens?

Reply:
https://github.com/VertaAI/modeldb/blob/master/client/verta/verta/client.py#L1380
This line in particular. In general, project and dataset, whichever is created after the other will re-initialize the context field of client.